services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/extensions/ChainExtension.java

```diff
@@ -112,25 +115,42 @@ public WriteModification beforeWrite(DynamoDbExtensionContext.BeforeWrite contex
             if (writeModification.transformedItem() != null) {
                 transformedItem = writeModification.transformedItem();
             }
-
-            if (writeModification.additionalConditionalExpression() != null) {
-                if (conditionalExpression == null) {
-                    conditionalExpression = writeModification.additionalConditionalExpression();
-                } else {
-                    conditionalExpression =
-                        Expression.join(conditionalExpression,
-                                        writeModification.additionalConditionalExpression(),
-                                        " AND ");
-                }
-            }
+            conditionalExpression = mergeConditionalExpressions(conditionalExpression, writeModification);
+            updateExpression = mergeUpdateExpressions(updateExpression, writeModification);
```

Comment:
At this point, I think this is a fairly specific behavior for ChainExtension (and it's also mostly a couple of if clauses). This may change when we achieve more clarity on the precedence of client vs request actions.